### PR TITLE
Private Review for LTR V2 Powershell Cmdlets

### PR DESCRIPTION
**Pull request by trgrie**
## Description

This is for an initial review, I'm not wanting to check this in, I just want to get some feedback so I'm ready when the time comes.

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#making-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+namespace Microsoft.Azure.Commands.Sql.Database_Backup.Cmdlet
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmSqlDatabaseLongTermRetentionBackup", DefaultParameterSetName = LocationNameSet, SupportsShouldProcess = true)]
+    public class GetAzureRmSqlDatabaseLongTermRetentionBackup : AzureSqlDatabaseLongTermRetentionBackupCmdletBase
+    {
+        /// <summary>
+        /// Parameter set name for backup name.
+        /// </summary>
+        private const string BackupNameSet = "BackupName";
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs]: In your cmdlet review you only had 6 parameter sets but here you added two more - why do you need these as separate parameter sets?  It seems that the first four could really just be two.

File: src/ResourceManager/Sql/Commands.Sql/StartupScripts/SqlStartup.ps1 [hunk 2]
```diff
@@ -18,6 +18,8 @@
     "Remove-AzureRmSqlDatabaseServerAuditing" = "Remove-AzureRmSqlServerAuditing";
     "Set-AzureRmSqlDatabaseServerAuditingPolicy" = "Set-AzureRmSqlServerAuditingPolicy";
     "Use-AzureRmSqlDatabaseServerAuditingPolicy" = "Use-AzureRmSqlServerAuditingPolicy";
+    "Get-AzureRmSqlDatabaseLongTermRetentionPolicy" = "Get-AzureRmSqlDatabaseBackupLongTermRetentionPolicy";
+    "Set-AzureRmSqlDatabaseLongTermRetentionPolicy" = "Set-AzureRmSqlDatabaseBackupLongTermRetentionPolicy";
 }.GetEnumerator() | Select @{Name='Name'; Expression={$_.Key}}, @{Name='Value'; Expression={$_.Value}} | `
 ForEach-Object {if (!(Get-Alias -Name $_.Name -ErrorAction Ignore)) {Set-Alias -Name $_.Name -Value $_.Value -Description "AzureAlias" -Scope Global}} | Out-Null
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/StartupScripts/SqlStartup.ps1]: Revert the changes to the file below this one.

File: src/ResourceManager/Sql/AzureRM.Sql.psd1 [hunk 3]
```diff
@@ -9,7 +9,7 @@
 @{
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/AzureRM.Sql.psd1]: Please update the markdown help using: https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md

Please add a section to the changelog here: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Sql/ChangeLog.md

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+
+        /// <summary>
+        /// The index of DatabaseName in the LTR Backup Resource ID.
+        /// </summary>
+        private const int DatabaseNameIndex = 9;
+
+        /// <summary>
+        /// The location the backups are in.
+        /// </summary>
+        [Parameter(Mandatory = true,
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs]: I don't think Location should be first, I think it should come after the other mandatory parameters.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+        /// <returns>The list of entities</returns>
+        protected override IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> GetEntity()
+        {
+            if (InputObject != null)
+            {
+                LocationName = InputObject.Location;
+                ServerName = InputObject.ServerName;
+                DatabaseName = InputObject.DatabaseName;
+            }
+            else if (!string.IsNullOrWhiteSpace(ResourceId))
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Instead of this, use ResourceIdentifier: https://github.com/Azure/azure-powershell/blob/9ee1941722db5a5d4d1bf8fa3630184389de08b3/src/ResourceManager/KeyVault/Commands.KeyVault/Models/PSVaultIdentityItem.cs#L31

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureSqlDatabaseBackupLongTermRetentionPolicy.cs [hunk 6]
```diff
@@ -21,23 +21,32 @@
...
 {
     [Cmdlet(VerbsCommon.Get, "AzureRmSqlDatabaseBackupLongTermRetentionPolicy", SupportsShouldProcess = true)]
+    [Alias("Get-AzureRmSqlDatabaseLongTermRetentionPolicy")]
     public class GetAzureSqlDatabaseBackupLongTermRetentionPolicy : AzureSqlDatabaseBackupLongTermRetentionPolicyCmdletBase
     {
+        /// <summary>
+        /// Gets or sets whether or not to use the Long Term Retention Vaults.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureSqlDatabaseBackupLongTermRetentionPolicy.cs]: Remove valuefrompipelinebypropertyname and position

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+using System.Management.Automation;
+using System.Globalization;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.Backup.Model;
+using Microsoft.Azure.Commands.Sql.Database.Model;
+
+namespace Microsoft.Azure.Commands.Sql.Database_Backup.Cmdlet
+{
+    [Cmdlet(VerbsCommon.Remove, "AzureRmSqlDatabaseLongTermRetentionBackup", DefaultParameterSetName = RemoveBackupDefaultSet, SupportsShouldProcess = true)]
+    public class RemoveAzureRmSqlDatabaseLongTermRetentionBackup : AzureSqlDatabaseLongTermRetentionBackupCmdletBase
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Remove cmdlets should implement switch parameter PassThru, where a user will get not output if not specified and either a boolean indicating failure/success or the object removed if the parameter is passed in: https://github.com/Azure/azure-powershell/blob/9ee1941722db5a5d4d1bf8fa3630184389de08b3/src/ResourceManager/KeyVault/Commands.KeyVault/Commands/RemoveAzureKeyVaultCertificate.cs#L62

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+            ValueFromPipeline = true,
+            HelpMessage = "The Resource ID of the Database Long Term Retention Backup to remove.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Defines whether it is ok to skip the requesting of rule removal confirmation
+        /// </summary>
+        [Parameter(HelpMessage = "Skip confirmation message for performing the action")]
+        public SwitchParameter Force { get; set; }
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs]: You only need force if this is a particularly destructive cmdlet that you really want the user to be sure about before running, or if there is an unexpected action that occurs because of it (i.e., a folder deletion will cause all files contained to be deleted).  Do you think this cmdlet qualifies?

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs [hunk 9]
```diff
@@ -0,0 +1,319 @@
...
+        /// No changes to persist to server
+        /// </summary>
+        /// <param name="entity">The output of apply user input to model</param>
+        /// <returns>The input entity</returns>
+        protected override IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> PersistChanges(
+            IEnumerable<AzureSqlDatabaseLongTermRetentionBackupModel> entity)
+        {
+            return entity;
+        }
+    }
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Should this cmdlet implement ExecuteCmdlet? Or does the base class cover the scenario entirely?

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+        public override void ExecuteCmdlet()
+        {
+            if (InputObject != null)
+            {
+                LocationName = InputObject.LocationName;
+                ServerName = InputObject.ServerName;
+                DatabaseName = InputObject.DatabaseName;
+                BackupName = InputObject.BackupName;
+            }
+            else if (!string.IsNullOrWhiteSpace(ResourceId))
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Use ResourceIdentifier here as well.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+            {
+                string[] tokens = ResourceId.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+
+                LocationName = tokens[LocationNameIndex];
+                ServerName = tokens[ServerNameIndex];
+                DatabaseName = tokens[DatabaseNameIndex];
+                BackupName = tokens[BackupNameIndex];
+            }
+
+            if (!Force.IsPresent && !ShouldProcess(
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Use this syntax: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Profile/Commands.Profile/Default/ClearAzureRmDefault.cs#L52

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Services/AzureSqlDatabaseBackupCommunicator.cs [hunk 12]
```diff
@@ -179,14 +188,105 @@ public Management.Sql.LegacySdk.Models.DeletedDatabaseBackup GetDeletedDatabaseB
...
+        /// </summary>
+        /// <param name="resourceGroup">The resource group name.</param>
+        /// <param name="serverName">The server name.</param>
+        /// <param name="databaseName">The database name.</param>
+        public Management.Sql.Models.LongTermRetentionPolicy GetDatabaseLongTermRetentionPolicy(
+            string resourceGroup,
+            string serverName,
+            string databaseName)
+        {
+            return GetCurrentSqlClient().LongTermRetentionPolicies.Get(resourceGroup, serverName, databaseName);
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/Database Backup/Services/AzureSqlDatabaseBackupCommunicator.cs]: How long will you need this "Current" vs "Legacy" syntax? Is that a permanent change?

File: src/ResourceManager/Sql/Commands.Sql/Database/Model/AzureSqlDatabaseModel.cs [hunk 13]
```diff
@@ -206,26 +206,26 @@ public AzureSqlDatabaseModel(string resourceGroup, string serverName, Management
...
+            MaxSizeBytes = database.MaxSizeBytes.Value;
             DatabaseName = database.Name;
             Status = database.Status;
             Tags = TagsConversionHelper.CreateTagDictionary(TagsConversionHelper.CreateTagHashtable(database.Tags), false);
             ElasticPoolName = database.ElasticPoolName;
             Location = database.Location;
             ResourceId = database.Id;
             CreateMode = database.CreateMode;
-            EarliestRestoreDate = database.EarliestRestoreDate;
+            EarliestRestoreDate = null; // database.EarliestRestoreDate;
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/Database/Model/AzureSqlDatabaseModel.cs]: Why are you making this change?

File: src/ResourceManager/Sql/Commands.Sql.Test/Commands.Sql.Test.csproj [hunk 14]
```diff
@@ -82,8 +82,8 @@
       <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.Azure.Management.Sql, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Sql.1.10.0-preview\lib\net452\Microsoft.Azure.Management.Sql.dll</HintPath>
-      <Private>True</Private>
+      <SpecificVersion>False</SpecificVersion>
```
[the remark this answers is not in the document]
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql.Test/Commands.Sql.Test.csproj]: That's the main reason I'm not ready to check in, the azure-sdk-for-net repo needs to be updated still.

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseBackupTests.cs [hunk 15]
```diff
@@ -104,6 +104,17 @@ public void TestRestoreLongTermRetentionBackup()
         }
         [Fact]
         [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestLongTermRetentionV2()
+        {
+            // TODO Rewrite SQL backup tests to be recordable
+            // TODO https://github.com/Azure/azure-powershell/issues/4155
```
[the remark this answers is not in the document]
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseBackupTests.cs]: It's hard since it takes ~18 hours to fully set up. The issue linked there explains that a bit.

File: src/ResourceManager/Sql/Commands.Sql/Database/Model/AzureSqlDatabaseModel.cs (hunk 13 above)
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database/Model/AzureSqlDatabaseModel.cs]: This has to do with the changes that @jaredmoo is making where some of these things can't be set correctly until the new sdk is updated.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Services/AzureSqlDatabaseBackupCommunicator.cs (hunk 12 above)
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database Backup/Services/AzureSqlDatabaseBackupCommunicator.cs]: No, not permanent. The Vaults are being decommissioned after about a month, so not long.

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseBackupTests.cs (hunk 15 above)
[the remark this answers is not in the document]
Comment (maddieclayton) [reply src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseBackupTests.cs]: Yikes.  Okay this is fine then, we'll leave the issue open to be figured out in the future.

File: src/ResourceManager/Sql/Commands.Sql/Database/Model/AzureSqlDatabaseModel.cs (hunk 13 above)
Comment (maddieclayton) [reply src/ResourceManager/Sql/Commands.Sql/Database/Model/AzureSqlDatabaseModel.cs]: So will this be reverted before you are ready to merge the PR?

**Comment by maddieclayton** [comment]
@trgrie I'm closing this PR - you should be able to see all of the comments still, and you can ping me if you have more questions.

File: src/ResourceManager/Sql/AzureRM.Sql.psd1 [hunk 16]
```diff
@@ -9,7 +9,7 @@
...
 
 # Script module or binary module file associated with this manifest.
-# RootModule = ''
+RootModule = '.\AzureRM.Sql.psm1'
```
[the remark this answers is not in the document]
Comment (trgrie) [reply src/ResourceManager/Sql/AzureRM.Sql.psd1]: Done.

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseBackupTests.ps1 [hunk 17]
```diff
@@ -145,6 +145,47 @@ function Test-RestoreLongTermRetentionBackup
...
+	$serverName = "trgrie-ltr-server"
+	$databaseName = "testdb"
+	$weeklyRetention1 = "P1W"
+	$weeklyRetention2 = "P2W"
+
+	# Basic Get Tests
+	Get-AzureRmSqlDatabaseLongTermRetentionBackup -Location $locationName
+	Get-AzureRmSqlDatabaseLongTermRetentionBackup -Location $locationName -ServerName $serverName
+	$backups = Get-AzureRmSqlDatabaseLongTermRetentionBackup -Location $locationName -ServerName $serverName -DatabaseName $databaseName
+	Get-AzureRmSqlDatabaseLongTermRetentionBackup -Location $locationName -ServerName $serverName -DatabaseName $databaseName -BackupName $backups[0].BackupName
```
[the remark this answers is not in the document]
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/DatabaseBackupTests.ps1]: Added.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs (hunk 1 above)
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs]: There may be a way to do this I couldn't find but it's tricky because if I combined all but the first out of the first four, there is a dependency that needs to happen (ie - if you provide server name, you need to provide location name). If there is a way to do that and it's better, I can make the change.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs (hunk 4 above)
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs]: I was following the order of the resource ID for LTR backups:
locations/longTermRetentionServers/longTermRetentionDatabases/longTermRetentionBackups

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs [hunk 18]
```diff
@@ -0,0 +1,319 @@
...
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "Whether or not to only get the latest backup per database. Defaults to false.")]
+        [Parameter(Mandatory = false,
+            ParameterSetName = GetBackupsByResourceIdSet,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "Whether or not to only get the latest backup per database. Defaults to false.")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter OnlyLatestPerDatabase { get; set; }
```
[the remark this answers is not in the document]
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Done.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs [hunk 19]
```diff
@@ -0,0 +1,319 @@
...
+            Position = 2,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The state of the database whose backups you want to find, Alive, Deleted, or All. Defaults to All")]
+        [Parameter(Mandatory = false,
+            ParameterSetName = GetBackupsByResourceIdSet,
+            Position = 2,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The state of the database whose backups you want to find, Alive, Deleted, or All. Defaults to All")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(new[] { Management.Sql.Models.DatabaseState.All, Management.Sql.Models.DatabaseState.Deleted, Management.Sql.Models.DatabaseState.Live },
```
[the remark this answers is not in the document]
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Done.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs (hunk 5 above)
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Done.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs (hunk 9 above)
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs]: No need here.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureSqlDatabaseBackupLongTermRetentionPolicy.cs (hunk 6 above)
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureSqlDatabaseBackupLongTermRetentionPolicy.cs]: Done. Also changed to current instead of legacy so we do not break existing workflows until LTR V1 is deprecated.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs (hunk 7 above)
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Added the PassThru Switch and I now override WriteResult and it is determined by the value of PassThru.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs (hunk 8 above)
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs]: I think so just because it could be a potential loss of data otherwise with a possibility of it being gone if the database has been dropped. That's my reasoning anyways.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs (hunk 10 above)
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Done.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs (hunk 11 above)
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Done. What does ShouldProcess do?

File: src/ResourceManager/Sql/Commands.Sql/Database/Model/AzureSqlDatabaseModel.cs (hunk 13 above)
Comment (trgrie) [reply src/ResourceManager/Sql/Commands.Sql/Database/Model/AzureSqlDatabaseModel.cs]: Correct. There are actually even more changes having to do with this now.

File: src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs (hunk 11 above)
Comment (maddieclayton) [reply src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Should process allows the user to specify the "Confirm" parameter, which will prompt them as soon as it hits this shouldprocess block.
